caffe2/serialize/inline_container.h

```diff
@@ -140,9 +171,14 @@ class CAFFE2_API PyTorchStreamReader final {
 
 class CAFFE2_API PyTorchStreamWriter final {
  public:
-  explicit PyTorchStreamWriter(std::string archive_name);
   explicit PyTorchStreamWriter(
-      const std::function<size_t(const void*, size_t)>& writer_func);
+      std::string archive_name,
+      const bool _write_version_at_setup=true);
```

Comment:
Not a huge fan of a flag here. Makes the behavior complicated. I feel like the version should just get written on finalization (seems like we already have a `finalized_` flag). And inline container should handle the maxing.

Reply:
Good idea. I wasn't a fan of it either but was leery of changing inline_container too much. Moving this to finalization should be nicer.